[chaaaaun] iP

chaaaaun opened the pull request:
# Plana

> “I will give you a million chances.” – Sensei

Plana frees your mind of having to remember things you need to do. It's,
- text-based
- _easy to learn_
- **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://github.com/chaaaaun/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a `Java` programmer, you can use it to practice `Java` too. Here's the `main` method:

```java
public static void main(String[] args) {
        try {
            Plana p = new Plana();
            p.init();
        } catch (ClassNotFoundException e) {
            System.out.println("Invalid save file:\n" + e.getMessage());
        } catch (IOException e) {
            System.out.println("Storage failed with error:\n" + e.getMessage());
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

chaaaaun committed:
Refactor and add greetings

chaaaaun committed:
Add colors and echo

chaaaaun committed:
Add record and list

chaaaaun committed:
Add exceptions

chaaaaun committed:
Add done functionality

chaaaaun committed:
Add different types of tasks

chaaaaun committed:
Add delete functionality

chaaaaun committed:
Add test cases

chaaaaun committed:
Fix warnings

chaaaaun committed:
Add save functionality

chaaaaun committed:
Add date parsing

chaaaaun committed:
Refactor code for OOP

chaaaaun committed:
Fix warnings

chaaaaun committed:
Merge remote-tracking branch 'origin/add-gradle-support'

chaaaaun committed:
Add gradle support

chaaaaun committed:
Add JUnit tests

chaaaaun committed:
Configure jar build task

chaaaaun committed:
Add Javadoc

chaaaaun committed:
Conform switch to coding standards

chaaaaun committed:
Add find functionality

chaaaaun committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/chatbot/Command.java

chaaaaun committed:
Merge branch 'branch-A-Javadoc'

# Conflicts:
#	src/main/java/chatbot/Command.java
#	src/main/java/chatbot/Task.java
#	src/main/java/chatbot/TaskList.java
#	src/main/java/chatbot/Ui.java

chaaaaun committed:
Add checkstyle

chaaaaun committed:
Correct codebase based on checkstyle

chaaaaun committed:
Fix regression

chaaaaun committed:
Update suppressions filter

kaitinghh commented on src/main/java/chatbot/Command.java:
```diff
@@ -0,0 +1,171 @@
+    EVENT("event"),
+    FIND("find");
+
+    private final String rep;
```
Final fields should be all capsed?

chaaaaun replied:
You're right, thanks for spotting!

kaitinghh commented on src/main/java/chatbot/Command.java:
```diff
@@ -0,0 +1,171 @@
+     * @throws DukeException If an error occurs with the execution.
+     */
+    private void executeAdd(Ui view, TaskList tl) throws DukeException {
+        Task t = new Task("");
```
What is the purpose of this line?

chaaaaun replied:
To initialise a default task, in case none of the switch cases below are applicable, should not happen in normal situations.

kaitinghh reviewed (commented):
LGTM!

chaaaaun committed:
Refactor for GUI

ondretann commented on src/main/java/chatbot/DeadlineTask.java:
```diff
@@ -0,0 +1,28 @@
+import chatbot.exceptions.InvalidArgumentException;
+
+class DeadlineTask extends Task implements Serializable {
+    private static final long serialVersionUID = 1L;
```
Will it be better if you caps it since its final? Otherwise, UID should not be in caps as well! But I do like how descriptive your constant is :)

ondretann commented on src/main/java/chatbot/Plana.java:
```diff
@@ -0,0 +1,43 @@
+public class Plana {
+    private final TaskList taskList;
+    private final Storage store;
+    private boolean shouldExit;
```
Will it be better to name this isExit instead to follow the java coding standard?

ondretann reviewed (commented):
Apart from minor errors that deviate from the java coding standard, everything else LGTM!

chaaaaun committed:
Fix warnings and code quality issues

There were some fields that were either unused
or had unclear naming.

Renaming these fields and removing them where
possible can improve code quality.

chaaaaun committed:
Add assertions and fix bugs

We are currently not using assertions in our code.

By using Java assertions we are able to signal that
we expect certain inputs into methods,
improving readability.

Let's use assertions to make our code more readable
and less error-prone.

chaaaaun committed:
Merge pull request #2 from chaaaaun/branch-A-CodeQuality

Fix warnings and code quality issues

chaaaaun committed:
Merge branch 'master' into branch-A-Assertions

chaaaaun committed:
Merge pull request #3 from chaaaaun/branch-A-Assertions

Add assertions and fix bugs

chaaaaun committed:
Fix regression and add CI

chaaaaun committed:
Fix workflow

chaaaaun committed:
Add unscheduled tasks with a fixed duration

chaaaaun committed:
Clean up GUI code

chaaaaun committed:
Remove traces of Duke

chaaaaun committed:
Add user guide

chaaaaun committed:
Update user guide

chaaaaun committed:
Fix regressions

chaaaaun committed:
Prep for release
